src/index.js

```diff
@@ -297,14 +300,17 @@ Multiaddr.prototype.equals = function equals (addr) {
  * // {family: 'IPv4', address: '127.0.0.1', port: '4001'}
  */
 Multiaddr.prototype.nodeAddress = function nodeAddress () {
-  if (!this.isThinWaistAddress()) {
-    throw new Error('Multiaddr must be "thin waist" address for nodeAddress.')
-  }
-
   const codes = this.protoCodes()
   const parts = this.toString().split('/').slice(1)
+
+  if (parts.length < 4 ||
+    (codes[0] !== 4 && codes[0] !== 41 && codes[0] !== 54 && codes[0] !== 55) ||
+    (parts[2] !== 'tcp' && parts[2] !== 'udp')) {
+    throw new Error('Multiaddr must be a valid address for nodeAddress.')
```

Comment:
Same goes here, should give the user an idea of what's not, not just that it's invalid.

Reply:
Agreed and fixed